Cannot find module 'hyper-dom-expressions' or its corresponding type declarations

### Describe the bug

When having `"skipLibCheck": false` in `tsconfig.json`, `solid-js/h` will throw the following error:

```
Cannot find module 'hyper-dom-expressions' or its corresponding type declarations
```

### Your Example Website or App

https://stackblitz.com/edit/vitejs-vite-noyuqg

### Steps to Reproduce the Bug or Issue

1. Go to https://stackblitz.com/edit/vitejs-vite-noyuqg
2. run `npm run build`


### Expected behavior

No error

### Screenshots or Videos

<img width="605" alt="image" src="https://github.com/solidjs/solid/assets/24715727/8c9b826e-d15c-4120-9993-01532c891c4d">


### Platform

- OS: [e.g. macOS, Windows, Linux]: N/A
- Browser: [e.g. Chrome, Safari, Firefox]: N/A
- Version: [e.g. 91.1]: N/A


### Additional context

_No response_

--- a/.changeset/beige-maps-notice.md
+++ b/.changeset/beige-maps-notice.md
@@ -1,0 +1,5 @@
+---
+"solid-js": patch
+---
+
+fix #1814 incorrect typing embedding for h and html

--- a/packages/solid/h/src/index.ts
+++ b/packages/solid/h/src/index.ts
@@ -1,4 +1,5 @@
 import { createHyperScript } from "./hyperscript";
+import type { HyperScript } from "./hyperscript";
 import {
   spread,
   assign,
@@ -8,7 +9,7 @@ import {
   SVGElements
 } from "solid-js/web";
 
-export default createHyperScript({
+const h: HyperScript = createHyperScript({
   spread,
   assign,
   insert,
@@ -16,3 +17,5 @@ export default createHyperScript({
   dynamicProperty,
   SVGElements
 });
+
+export default h;

--- a/packages/solid/html/src/index.ts
+++ b/packages/solid/html/src/index.ts
@@ -1,4 +1,5 @@
 import { createHTML } from "./lit";
+import type { HTMLTag } from "./lit";
 import {
   effect,
   style,
@@ -22,7 +23,7 @@ import {
   SVGNamespace
 } from "solid-js/web";
 
-export default createHTML({
+const html: HTMLTag = createHTML({
   effect,
   style,
   insert,
@@ -44,3 +45,5 @@ export default createHTML({
   SVGElements,
   SVGNamespace
 });
+
+export default html;

